ibis/omniscidb/tests/test_client.py

```diff
@@ -287,3 +287,20 @@ def test_cpu_execution_type(
 
     for mocked_method in mocked_methods:
         mocked_method.stop()
+
+
+def test_cpu_ipc_execution_mode(con):
+    con.execution_mode('cpu')
+    result = con.table('functional_alltypes').head().execute()
+    assert isinstance(result, pd.DataFrame)
+
+    try:
+        con.execution_mode('gpu')
+    except Exception as e:
+        assert e.error_msg == (
+            'Cannot switch to GPU mode in a'
+            ' server started in CPU-only mode.'
+        )
+
+    with pytest.raises(ValueError):
```

Comment:
In general it's preferred to have three tests for this instead of one, one for cpu, one for gpu, and a different one for an invalid error. If you name the tests properly, then when there is a failure, you know exactly why it's failing without having to go into the details of the test.

Reply:
sounds good. I will work on that. thanks!